### PR TITLE
Botan 3.2.0: Prepare Documents for Submission

### LINKED ISSUE #108: Move Method Description of DATA-based Side Channel Analysis to Audit Method Document
The high-level DATA-based methodology is currently documented in the "Audit Report" document. It might be better to move this into the "Audit Method" document.

Its probably fine to do that in the next iteration (i.e. [Botan 3.2.0](https://github.com/randombit/botan/milestone/5) due in October).

### PR DESCRIPTION
**Pull request by reneme**
Botan 3.2.0 was released today (09th October 2023). This adds final touches to the documents prior to the submission. 
This includes:

* Some 3.2.0-specific introductory and summarizing remarks in the `audit_report` document
* Incorporation of preliminary updates from #138 (now obsolete)
* Moving DATA methodology description into the `audit_method` document (#108)
* Auxiliary updates of documents (change logs, authorships, configs, "audited modules" list, ...)

Probably it makes sense to go through the individual commits of this pull request.

closes #108.

### CODE REVIEW

File: docs/audit_report/src/00_09_introduction.rst [hunk 1]
```diff
@@ -50,162 +50,167 @@ unit and integration tests (in ``src/tests``, ``src/bogo_shim``,
...
+``aes_power8``, ``aes_vperm``, ``argon2_avx2``, ``argon2_ssse3``,
+``certstor_flatfile``, ``certstor_sql``, ``certstor_sqlite3``,
+``certstor_system``, ``certstor_system_macos``, ``certstor_system_windows``,
+``dilithium``, ``dilithium_aes``, ``ffi``, ``ghash_cpu``, ``ghash_vperm``,
+``keccak_perm_bmi2``, ``kyber``, ``kyber_90s``, ``pkcs11``, ``sha1_armv8``,
+``sha1_sse2``, ``sha1_x86``, ``sha2_32_armv8``, ``sha2_32_bmi2``,
+``sha2_32_x86``, ``sha2_64_bmi2``, ``shake``, ``sphincsplus_sha2``,
+``sphincsplus_shake``, ``tls_cbc``, ``tls12``, ``tls13``, ``tls13_pqc``,
+``xts``. Patches that don't alter any of the above-mentioned components or
+relevant modules are considered out-of-scope.
```
Comment (FAlbertDev) [inline docs/audit_report/src/00_09_introduction.rst:66]: This list of additional modules also contains the `if_available` modules of the BSI policy, like `aes_armv8`, for example. We do not want that, do we?

File: docs/audit_report/src/01_generic_changes.rst [hunk 2]
```diff
@@ -1,6 +1,59 @@
...
+| PQ KEM                         | Classical Key Exchange         |
++--------------------------------+--------------------------------+
+| | Kyber R3 512                 | | X25519                       |
+| | Kyber R3 512                 | | NIST P-256                   |
+| | Kyber R3 768                 | | X25519                       |
+| | Kyber R3 768                 | | NIST P-384                   |
+| | Kyber R3 1024                | | NIST P-1024                  |
++--------------------------------+--------------------------------+
+
+Additionally, handshakes with just Kyber as a PQ KEM and without any classical
```
Comment (FAlbertDev) [inline docs/audit_report/src/01_generic_changes.rst:24]: image

In comparison to other tables this one looks somehow weird. Maybe we want to add row separators.

File: docs/audit_report/src/01_generic_changes.rst [hunk 3]
```diff
@@ -1,6 +1,59 @@
...
+With ``XOF``, Botan now has a base-class API (cf. ``HashFunction`` or
+``BlockCipher``) to support eXtendable Output Functions (XOF). The public API provides
+"SHAKE-128" and "SHAKE-256" as XOFs.
+
+Kuznyechik
+~~~~~~~~~~
+
+The National Standard of the Russian Federation GOST R 34.12-2015 defines
+Kuznyechik as a block cipher with a block length of 128 bits. This algorithm is
+_prohibited_ in the "BSI" build policy.
```
Comment (FAlbertDev) [inline docs/audit_report/src/01_generic_changes.rst:53]: Do we need a reference for Kuznyechik? Also, `_prohibited_` is not rendered correctly:
image

File: docs/audit_report/src/02_security_issues.rst [hunk 4]
```diff
@@ -1,7 +1,5 @@
 Security and Vulnerabilities
 ============================
```
Comment (FAlbertDev) [inline docs/audit_report/src/02_security_issues.rst:3]: Not here, but in the rendered audit_report page 13: 4.2.3 Relevant Patches. The table seem to overflow the footer:
image

Also, there is an unnecessary blank page before this table.

**Review by FAlbertDev — COMMENTED** [review]
All in all LGTM. Only some optics and nits.

File: docs/audit_report/src/01_generic_changes.rst (hunk 2 above)
Comment (reneme) [reply docs/audit_report/src/01_generic_changes.rst:24]: Yeah. Good idea. Done.

File: docs/audit_report/src/01_generic_changes.rst (hunk 3 above)
Comment (reneme) [reply docs/audit_report/src/01_generic_changes.rst:53]: Done. Linked it to [RFC7801](https://www.rfc-editor.org/rfc/rfc7801.html).

File: docs/audit_report/src/02_security_issues.rst (hunk 4 above)
Comment (reneme) [reply docs/audit_report/src/02_security_issues.rst:3]: Oh crap.

File: docs/audit_report/scripts/audited_modules_list.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+
+def print_table(columns: int, modules: list[str]):
+    print(".. list-table::")
+    print()
+
+    i = 0
+    for module in modules:
+        if(i == 0):
+            print("   * - {}".format(module))
```
[the remark this answers is not in the document]
Comment (reneme) [reply docs/audit_report/scripts/audited_modules_list.py:75]: Yeah, and no parenthesis around the `if` condition. 🤣

# 😛

**Comment by reneme** [comment]
[Fixed your remarks](https://github.com/sehlen-bsi/botan-docs/pull/143/commits/dce6ccb067263de57177ea0e81b79d95676559b2), except the overflowing table. Yikes, I don't want to workaround that. We should all just get used to HTML. 😭

File: docs/audit_report/src/02_security_issues.rst (hunk 4 above)
Comment (reneme) [reply docs/audit_report/src/02_security_issues.rst:3]: [Got it](https://github.com/sehlen-bsi/botan-docs/pull/143/commits/2941b1a021e4f06e37fc1cf113e250fd24e04298)! The Sphinx documentation states that it automatically renders tables with more than 30 rows as "longtable". Longtables are allowed to break across pages (and we've seen that before in the 3.0.0 document). Unfortunately, this particular table only had 27 rows.

Now I forced all of those patch tables to be longtables.

**Review by FAlbertDev — APPROVED** [review]
🚀

File: docs/audit_report/src/00_09_introduction.rst (hunk 1 above)
Comment (lieser) [reply docs/audit_report/src/00_09_introduction.rst]: I think the following (platform dependent) modules should still be in this list, as they are not in the BSI policy nor a requirement of any of the modules:

- `certstor_system_macos`
- `certstor_system_windows`
- `certstor_sqlite3`
- `sha1_armv8`
- `sha1_sse2`
- `sha1_x86`

Basically all modules that we review but that would need to be explicitly mentioned in `--enable-modules=` to be enabled.

Comment (reneme) [reply docs/audit_report/src/00_09_introduction.rst]: Mhm, fair enough. I'll open a pull request for those.

**Pull request merged**